mmdet3d/core/bbox/coders/monoflex_bbox_coder.py

```diff
@@ -0,0 +1,387 @@
+
+        Return:
+            tuple(torch.Tensor): Centers of 3D boxes.
+                shape: (B * max_objs, 3)
+        """
+        # (B * max_objs, 10, 2) 8 projected corners,
+        # top center and bottom center
+        pred_keypoints2d = pred_keypoints2d.view(-1, 10, 2)
+
+        N = pred_keypoints2d.shape[0]
+        # (B * max_objs, 3)
+        pred_height_3d = pred_dimensions[:, 1].clone()
+        # (B, 4, 4)
+        N_batch = cam2imgs.shape[0]
+        batch_id = torch.arange(N_batch).unsqueeze(1)
+        obj_id = batch_id.repeat(1, N // N_batch).flatten()
+        # (B * max_objs, 4, 4)
+        cam2imgs = cam2imgs[obj_id]
+        # (B * max_objs, )
+        f_u = cam2imgs[:, 0, 0]
+        # (B * max_objs, )
+        center_height = pred_keypoints2d[:, -2, 1] - pred_keypoints2d[:, -1, 1]
+        # (B * max_objs, 2)
+        corner_02_height = \
+            pred_keypoints2d[:, [0, 2], 1] - pred_keypoints2d[:, [4, 6], 1]
+        corner_13_height = \
+            pred_keypoints2d[:, [1, 3], 1] - pred_keypoints2d[:, [5, 7], 1]
+
+        center_depth = f_u * pred_height_3d / (
+            F.relu(center_height) * down_ratio + self.eps)
+        corner_02_depth = f_u * pred_height_3d / (
+            F.relu(corner_02_height) * down_ratio + self.eps)
+        corner_13_depth = f_u * pred_height_3d / (
+            F.relu(corner_13_height) * down_ratio + self.eps)
+        corner_02_depth = corner_02_depth.mean(dim=1)
+        corner_13_depth = corner_13_depth.mean(dim=1)
+
+        pred_keypoints_depth = torch.stack(
+            (center_depth, corner_02_depth, corner_13_depth), dim=1)
+        # (B*max_objs, 3)
+        pred_keypoints_depth = torch.clamp(
+            pred_keypoints_depth,
+            min=self.depth_range[0],
+            max=self.depth_range[1])
+
+        return pred_keypoints_depth
+
+    def decode_dimension(self, labels, dims_offset):
+        """Retrieve object dimensions.
+
+        Args:
+            labels (torch.Tensor): Each points' category id.
+                shape (B* max_objs, K)
+            dims_offset (torch.Tensor): Dimension offsets.
+                shape (B* max_objs, 3)
+
+        Returns:
+            torch.Tensor: Shape (N, 3)
+        """
+        labels = labels.flatten().long()
+        dims_std = dims_offset.new_tensor(self.dims_std)
+        dims_mean = dims_offset.new_tensor(self.dims_mean)
+        cls_dimension_mean = dims_mean[labels, :]
+
+        if self.dims_modes[0] == 'exp':
+            dims_offset = dims_offset.exp()
+
+        if self.dims_modes[2]:
+            cls_dimension_std = dims_std[labels, :]
+            dimensions = dims_offset * cls_dimension_std + cls_dimension_mean
+        else:
+            dimensions = dims_offset * cls_dimension_mean
+
+        return dimensions
+
+    def decode_orientation(self, vector_ori, locations):
+        """Retrieve object orientation.
+
+        Args:
+            vector_ori(torch.Tensor): Local vector orientation
```

Comment:
add a space between 'vector_ori' and '(torch.Tensor)' and 'locations' and '(torch.Tensor)'